swift/ql/test/library-tests/dataflow/taint/core/conversions.swift

```diff
@@ -23,68 +29,111 @@ class MyString : LosslessStringConvertible, CustomStringConvertible, CustomDebug
 
 func testConversions() {
-	sink(arg: sourceInt()) // $ tainted=26
-	sink(arg: Int(sourceInt())) // $ tainted=27
-	sink(arg: UInt8(sourceInt())) // $ tainted=28
-	sink(arg: Float(sourceInt())) // $ tainted=29
-	sink(arg: String(sourceInt())) // $ tainted=30
-	sink(arg: String(sourceInt()).utf8) // $ tainted=31
+	sink(arg: sourceInt()) // $ tainted=32
+	sink(arg: Int(sourceInt())) // $ tainted=33
+	sink(arg: UInt8(sourceInt())) // $ tainted=34
+	sink(arg: Float(sourceInt())) // $ tainted=35
+	sink(arg: String(sourceInt())) // $ tainted=36
+	sink(arg: String(sourceInt()).utf8) // $ tainted=37
 
 	let arr = [1, 2, sourceInt()]
 	sink(arg: arr)
-	sink(arg: arr[0]) // $ tainted=33
+	sink(arg: arr[0]) // $ tainted=39
 	sink(arg: [MyInt](arr))
-	sink(arg: [MyInt](arr)[0]) // $ tainted=33
+	sink(arg: [MyInt](arr)[0]) // $ tainted=39
 	sink(arg: [UInt8](sourceString().utf8))
-	sink(arg: [UInt8](sourceString().utf8)[0]) // $ tainted=39
+	sink(arg: [UInt8](sourceString().utf8)[0]) // $ tainted=45
 
 	if let v = sourceInt() as? UInt {
-		sink(arg: v) // $ tainted=41
+		sink(arg: v) // $ tainted=47
 	}
 
 	let v2: UInt8 = numericCast(sourceInt())
-	sink(arg: v2) // $ tainted=45
+	sink(arg: v2) // $ tainted=51
 
 	let v4: UInt = unsafeBitCast(sourceInt(), to: UInt.self)
-	sink(arg: v4) // $ tainted=48
+	sink(arg: v4) // $ tainted=54
 
 	let v5 = UInt(truncatingIfNeeded: sourceInt())
-	sink(arg: v5) // $ tainted=51
+	sink(arg: v5) // $ tainted=57
 
 	let v6 = UInt(bitPattern: sourceInt())
-	sink(arg: v6) // $ tainted=54
+	sink(arg: v6) // $ tainted=60
 
-	sink(arg: Int(exactly: sourceInt())!) // $ tainted=57
-	sink(arg: Int(clamping: sourceInt())) // $ tainted=58
-	sink(arg: Int(truncatingIfNeeded: sourceInt())) // $ tainted=59
-	sink(arg: Int(sourceString(), radix: 10)!) // $ tainted=60
+	let v7 = abs(sourceInt())
+	sink(arg: v7) // $ tainted=63
 
-	sink(arg: Int(littleEndian: sourceInt())) // $ tainted=62
-	sink(arg: Int(bigEndian: sourceInt())) // $ tainted=63
-	sink(arg: sourceInt().littleEndian) // $ tainted=64
-	sink(arg: sourceInt().bigEndian) // $ tainted=65
+	let v8 = UInt64(0)
+	sink(arg: v8)
+	sink(arg: v8.advanced(by: 1))
+	sink(arg: v8.advanced(by: sourceInt())) // $ tainted=69
 
-	// ---
+	sink(arg: Int(exactly: sourceInt())!) // $ tainted=71
+	sink(arg: UInt32(exactly: sourceInt())!) // $ tainted=72
+	sink(arg: Int(clamping: sourceInt())) // $ tainted=73
+	sink(arg: Int(truncatingIfNeeded: sourceInt())) // $ tainted=74
+	sink(arg: Int(sourceString(), radix: 10)!) // $ tainted=75
+
+	sink(arg: Int(littleEndian: sourceInt())) // $ tainted=77
+	sink(arg: Int(bigEndian: sourceInt())) // $ tainted=78
+	sink(arg: sourceInt().littleEndian) // $ tainted=79
+	sink(arg: sourceInt().bigEndian) // $ tainted=80
+
+	let (q1, r1) = 1000.quotientAndRemainder(dividingBy: 2)
+	sink(arg: q1)
+	sink(arg: r1)
+
+	let (q2, r2) = sourceInt().quotientAndRemainder(dividingBy: 2)
+	sink(arg: q2) // $ MISSING: tainted=86
+	sink(arg: r2) // $ MISSING: tainted=86
```

Comment:
The test cases involving returned tuples are failing.  I've planned investigating this as future work, unless someone can spot out what's wrong and maybe how to fix it?

Reply:
Does the [0..1] syntax work for tuples?

Reply:
Good question.  As far as I can tell (from examining the `AccessPathToken` class) it's not supported, though the syntax `[0,1]` is so I shall probably switch to that.  It's not the only problem though - `addingReportingOverflow` doesn't use this syntax and we don't get flow through that either.

Reply:
I've pushed this change, and created an issue for the remaining flow failure.